CONTRIBUTING.md

```diff
@@ -0,0 +1,278 @@
+to mark a cell for `nbval` to change its testing behavior.
+For more information have a look at `nbval`'s documentation
+[Avoid output comparison for specific cells](https://nbval.readthedocs.io/en/latest/index.html#Avoid-output-comparison-for-specific-cells)
+and
+[Using tags instead of comments](https://nbval.readthedocs.io/en/latest/index.html#Using-tags-instead-of-comments)
+.
+
+### flake8-nb
+
+As for `nbval` we also encourage to use
+[cell tags](https://flake8-nb.readthedocs.io/en/latest/usage.html#cell-tags)
+to configure the reported code style violations of `flake8-nb`.
+Please only use this scarcely and when absolutely needed,
+i.e. if you have a cell with different language code or if you want to showcase
+bad code.
+
+## Add your changes to the docs
+
+To make the provided information more accessible (i.e. on mobile when you are on your way to university),
+we also generate documentation as an html page, PDF and epub, which is published at
+[Read The Docs](https://readthedocs.org/projects/python-tools-for-students).
+
+### Adding notebooks
+
+Notebooks are included in the docs using
+[`nbsphinx`](https://nbsphinx.readthedocs.io/)
+and
+[`nbsphinx-link`](https://github.com/vidartf/nbsphinx-link).
+In order to add a notebook to the docs, you need to create a `*.nblink`
+file in the appropriate folder in the docs and add its path to
+`docs/material.rst` / `docs/examples.rst`.
+If your notebooks contain extra media like images,
+you need to add them as `extra-media` entry in the `*.nblink` file.
+
+### Adding markdown files
+
+Markdown files are included in the docs using
+[`recommonmark`](https://recommonmark.readthedocs.io/).
+Sadly `recommonmark` does not recognize files outside of the docs
+root folder (`docs`). So in order not to copy files and maintain two versions,
+the best solution is to use `symbolic links` inside the `docs` folder
+which point to the appropriate file.
+In order for this to work for all operating systems, you should add
+a new line to `docs/refresh_symlinks.sh`,
+and run it to create a `symbolic link` in the `docs` folder.
+After that you should add the `symlink` to `.gitattributes` as `symlink=file`.
+After that you can include it in any `*.rst` file
+[as you would normally](https://www.sphinx-doc.org/en/master/usage/quickstart.html).
+
+### Building the docs locally
+
+To build the documentation, open a terminal, navigate to the `docs` folder and
+run `make html` (Posix like) / `make.bat html` (Windows).
+This will create the documentation inside the folder `docs/_build/html`.
+
+<div class="admonition note">
+
+<p class="admonition-title">Note:</p>
+<p>
+For the docs to be build it is mandatory that you use a conda installation of python or at least have conda installed.
+This is due to the fact that the notebook inclusion in the docs utilizes the tool [pandoc](https://pandoc.org/).
+Even so `pandoc` is present in many package manager repository indices, this is mostly an too old version,
```

Comment:
typo: "... this is mostly a too old version"

Reply:
fixed